### PR TITLE
Gerudo2 OK and mostly documented

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Purple-clad Gerudo

All functions matched, all variables named, enums and defines. I haven't bothered with enums for limbs since it's basically only the head that is referenced.

Similar to EnGe3, with all the walking and catching actions as well. I don't completely understand how these work together, although individually they're way simpler than the Heishi ones, mostly just distance checks in the field of view.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 1]
```diff
@@ -24,65 +62,598 @@ const ActorInit En_Ge2_InitVars = {
...
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_8002E4B4(globalCtx, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+
+    if ((!(this->stateFlags & GE2_STATE_ANIMCOMPLETE)) && SkelAnime_Update(&this->skelAnime)) {
+        this->stateFlags |= GE2_STATE_ANIMCOMPLETE;
+    }
+}
+
+// Same as EnGe3_MoveAndBlink
+void EnGe3_MoveAndBlink(EnGe2* this, GlobalContext* globalCtx) {
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:498]:
```suggestion
// Same as EnGe3_MoveAndBlink
void EnGe2_MoveAndBlink(EnGe2* this, GlobalContext* globalCtx) {
```
Might be the same, but probably shouldn't be named the same.

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 2]
```diff
@@ -24,65 +62,598 @@ const ActorInit En_Ge2_InitVars = {
...
+    }
+
+    this->stateFlags = 0;
+    this->unk_304 = 0; // Set and not used
+    this->walkTimer = 0;
+    this->playerSpottedParam = 0;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.gravity = -1.0f;
+    this->walkDirection = this->actor.posRot.rot.y;
+    this->walkDuration = ((this->actor.params & 0xFF00) >> 8) * 0xA;
```
Comment (petrie911) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:156]:
```suggestion
    this->walkDuration = ((this->actor.params & 0xFF00) >> 8) * 10;
```

**Review by fig02 — APPROVED** [review]
changes made

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.h [hunk 3]
```diff
@@ -6,9 +6,28 @@
...
+
 typedef struct EnGe2 {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x1C0];
+    /* 0x014C */ ColliderCylinder collider;
+    /* 0x0198 */ SkelAnime skelAnime;
+    /* 0x01DC */ Vec3s jointTable[22];
+    /* 0x0260 */ Vec3s morphTable[22];
+    /* 0x02E4 */ s16 eyeIndex; // eyeIndex
+    /* 0x02E6 */ s16 blinkTimer; // blinkTimer
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.h:18]:
```suggestion
    /* 0x02E4 */ s16 eyeIndex;
    /* 0x02E6 */ s16 blinkTimer;
```

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 4]
```diff
@@ -1,18 +1,56 @@
...
+#include "vt.h"
 
 #define FLAGS 0x00000019
 
 #define THIS ((EnGe2*)thisx)
 
+#define GE2_STATE_ANIMCOMPLETE 1 << 1
+#define GE2_STATE_KO 1 << 2
+#define GE2_STATE_CAPTURING 1 << 3
+#define GE2_STATE_TALKED 1 << 4
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:12]:
```suggestion
#define GE2_STATE_ANIMCOMPLETE (1 << 1)
#define GE2_STATE_KO (1 << 2)
#define GE2_STATE_CAPTURING (1 << 3)
#define GE2_STATE_TALKED (1 << 4)
```

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 5]
```diff
@@ -44,65 +82,582 @@ static ColliderCylinderInit D_80A34310 = {
...
+    0x060098AC,
+    0x060011F4, // Falling to ground
+    0x060098AC,
+    0x06008D60, // Runnng, spear in attack position
+    0x060098AC, 0x060098AC, 0x060098AC,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge2/EnGe2_Update.s")
+static u8 sAnimModes[] = { ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP,
+                           ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE };
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:103]:
```suggestion
                           ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, };
```

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 6]
```diff
@@ -44,65 +82,582 @@ static ColliderCylinderInit D_80A34310 = {
...
+
+extern FlexSkeletonHeader D_06008968;
+extern AnimationHeader D_06009ED4;
+
+void EnGe2_ChangeAction(EnGe2* this, s32 i) {
+    this->actionFunc = sActionFuncs[i];
+    Animation_Change(&this->skelAnime, sAnimations[i], 1.0f, 0.0f, Animation_GetLastFrame(sAnimations[i]),
+                     sAnimModes[i], -8.0f);
+    this->stateFlags &= ~(GE2_STATE_ANIMCOMPLETE);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:112]:
```suggestion
    this->stateFlags &= ~GE2_STATE_ANIMCOMPLETE;
```

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 7]
```diff
@@ -44,65 +82,582 @@ static ColliderCylinderInit D_80A34310 = {
...
+    }
+}
+
+void EnGe2_TurnPlayerSpotted(EnGe2* this, GlobalContext* globalCtx) {
+    s32 playerSpotted;
+
+    this->actor.speedXZ = 0.0f;
+
+    if (this->stateFlags & GE2_STATE_TALKED) {
+        this->stateFlags &= ~(GE2_STATE_TALKED);
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:315]:
```suggestion
        this->stateFlags &= ~GE2_STATE_TALKED;
```

File: src/overlays/actors/ovl_En_Ge2/z_en_ge2.c [hunk 8]
```diff
@@ -44,65 +82,582 @@ static ColliderCylinderInit D_80A34310 = {
...
+
+void EnGe2_MaintainColliderAndSetAnimState(EnGe2* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+
+    if ((!(this->stateFlags & GE2_STATE_ANIMCOMPLETE)) && SkelAnime_Update(&this->skelAnime)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ge2/z_en_ge2.c:494]:
```suggestion
    if (!(this->stateFlags & GE2_STATE_ANIMCOMPLETE) && SkelAnime_Update(&this->skelAnime)) {
```

**Comment by fig02** [comment]
changes made

**Review by Roman971 — APPROVED** [review]

**Pull request merged**